File: client/src/model/Construction.ts
import ConstructionType from './ConstructionType'
import ConstructionSubtype from './ConstructionSubtype'
import WeldingControl from './WeldingControl'

interface IConstruction {
	id: number
	name: string
	type: ConstructionType
	subtype: ConstructionSubtype
	valuation: string
	standardAlbumCode: string
	numOfStandardConstructions: number
	paintworkCoeff: number
	weldingControl: WeldingControl
	hasEdgeBlunting: boolean
	hasDynamicLoad: boolean
	hasFlangedConnections: boolean
}

class Construction {
	id: number
	name: string
	type: ConstructionType
	subtype: ConstructionSubtype
	valuation: string
	standardAlbumCode: string
	numOfStandardConstructions: number
	paintworkCoeff: number
	weldingControl: WeldingControl
	hasEdgeBlunting: boolean
	hasDynamicLoad: boolean
	hasFlangedConnections: boolean

	constructor(obj?: IConstruction) {
		this.id = (obj && obj.id) || 0
		this.name = (obj && obj.name) || ''
		this.type = (obj && obj.type) || null
		this.subtype = (obj && obj.subtype) || null
		this.valuation = (obj && obj.valuation) || ''
		this.standardAlbumCode = (obj && obj.standardAlbumCode) || ''
		this.numOfStandardConstructions =
			(obj && obj.numOfStandardConstructions) || 0
		this.paintworkCoeff = (obj && obj.paintworkCoeff) || 0
		this.weldingControl = (obj && obj.weldingControl) || null
		this.hasEdgeBlunting = (obj && obj.hasEdgeBlunting) || false
		this.hasDynamicLoad = (obj && obj.hasDynamicLoad) || false
		this.hasFlangedConnections = (obj && obj.hasFlangedConnections) || false
	}
}

export default Construction

File: server/DocumentsKM/Models/Construction.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Вид конструкции
    public class Construction
    {
        [Key]
        public int Id { get; set; }

        // Выпуск спецификации
        [Required]
        [ForeignKey("SpecificationId")]
        public virtual Specification Specification { get; set; }

        // Наименование
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        // Вид
        [Required]
        [ForeignKey("TypeId")]
        public virtual ConstructionType Type { get; set; }

        // Подвид
        [ForeignKey("SubtypeId")]
        public virtual ConstructionSubtype Subtype { get; set; }
        public int? SubtypeId { get; set; }

        // Расценка
        [MaxLength(10)]
        public string Valuation { get; set; }

        // Шифр типового альбома
        [MaxLength(20)]
        public string StandardAlbumCode { get; set; }

        // Количество типовых конструкций
        [Required]
        [Range(0, 1000000, ErrorMessage = "Value should be greater than or equal to 0")]
        public int NumOfStandardConstructions { get; set; }

        // Притупление кромок
        [Required]
        public bool HasEdgeBlunting { get; set; }

        // Динамическая нагрузка
        [Required]
        public bool HasDynamicLoad { get; set; }

        // Фланцевые соединения
        [Required]
        public bool HasFlangedConnections { get; set; }

        // Контроль плотности сварных швов
        [Required]
        [ForeignKey("WeldingControlId")]
        public virtual WeldingControl WeldingControl { get; set; }

        // Коэффициент окрашивания
        [Required]
        [Range(0, 1000000, ErrorMessage = "Value should be greater than or equal to 0")]
        public float PaintworkCoeff { get; set; }

        // Элементы вида конструкции
        public virtual IList<ConstructionElement> ConstructionElements { get; set; } = new List<ConstructionElement>();

        // Высокопрочные болты
        public virtual IList<ConstructionBolt> ConstructionBolts { get; set; } = new List<ConstructionBolt>();
    }
}
